### ISSUE TITLE
[JENKINS-24295] Gerrit Cancel doesn't work for Matrix jobs

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/matrix-project-plugin#584]
<p>The "Build Current Patches Only" feature works great for all normal jobs. But matrix jobs are not canceled when they are still running and a new patch is uploaded.</p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=rschulz">rschulz</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-24295" target="_blank">Gerrit Cancel doesn't work for Matrix jobs</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/rsandell.png" title="rsandell's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=rsandell">rsandell</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Major
<li><b>component(s)</b>: gerrit-trigger-plugin, matrix-project-plugin
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2014-09-11T09:26:19+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 3
<li><b>imported</b>: 2025-12-06
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>The "Build Current Patches Only" feature works great for all normal jobs. But matrix jobs are not canceled when they are still running and a new patch is uploaded.</p></pre>
</details>
</details>
<ul><li><i>environment</i>: <code>Linux, Jenkins 1.552, Gerrit Trigger 2.11.0</code></li></ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>rschulz:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/24295?focusedId=208044&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-208044">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-24295?focusedId=208044&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-208044">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>It seems to work correctly for matrix jobs as long as all the sub-jobs have been started at the time the patch is uploaded. But if some sub-jobs are still waiting on an executor to become available, than those waiting jobs are not deleted from the queue.</p>
</pre>
</details>

</ul>
</details>

<p>It seems to work correctly for matrix jobs as long as all the sub-jobs have been started at the time the patch is uploaded. But if some sub-jobs are still waiting on an executor to become available, than those waiting jobs are not deleted from the queue.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/rin_ne.png" title="rin_ne's avatar" /> rin_ne:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/24295?focusedId=210301&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-210301">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-24295?focusedId=210301&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-210301">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I can reproduce then find root cause. New PR is created.</p>
</pre>
</details>

</ul>
</details>

<p>I can reproduce then find root cause. New PR is created.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/24295?focusedId=210307&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-210307">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-24295?focusedId=210307&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-210307">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: rinrinne
Path:
 src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritTrigger.java
<a href="http://jenkins-ci.org/commit/gerrit-trigger-plugin/8b0e9be7c5be6c70c53aeb25c8a1a258ac3d54da" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/gerrit-trigger-plugin/8b0e9be7c5be6c70c53aeb25c8a1a258ac3d54da</a>
Log:
  Add one-off executor to search list for cancel job</p>

<p>By "Build Current Pathcset Only" feature, running and queued jobs with
the same change are canceled. These are found in queue items and
executors.</p>

<p>But MatrixJob uses one-off executor. It is not included in executors. So
it cannot be canceled.</p>

<p>This patch adds one-off executors to executor's list.</p>

<p>Fix for <a href="https://issue-redirect.jenkins.io/issue/24295" title="Gerrit Cancel doesn't work for Matrix jobs" class="issue-link" data-issue-key="JENKINS-24295"><del>JENKINS-24295</del></a></p>

<p>Task-Url: <a href="https://issues.jenkins-ci.org/browse/JENKINS-24295" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-24295</a></p>


</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: rinrinne<br/>
Path:<br/>
 src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritTrigger.java<br/>
<a href="http://jenkins-ci.org/commit/gerrit-trigger-plugin/8b0e9be7c5be6c70c53aeb25c8a1a258ac3d54da" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/gerrit-trigger-plugin/8b0e9be7c5be6c70c53aeb25c8a1a258ac3d54da</a><br/>
Log:<br/>
  Add one-off executor to search list for cancel job</p>

<p>By "Build Current Pathcset Only" feature, running and queued jobs with<br/>
the same change are canceled. These are found in queue items and<br/>
executors.</p>

<p>But MatrixJob uses one-off executor. It is not included in executors. So<br/>
it cannot be canceled.</p>

<p>This patch adds one-off executors to executor's list.</p>

<p>Fix for <a href="https://issue-redirect.jenkins.io/issue/24295" title="Gerrit Cancel doesn't work for Matrix jobs" class="issue-link" data-issue-key="JENKINS-24295"><del>JENKINS-24295</del></a></p>

<p>Task-Url: <a href="https://issues.jenkins-ci.org/browse/JENKINS-24295" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-24295</a></p>
